src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,43 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+
+        private boolean isPrivate;
+        public String value;
+        
+        public Contact(String contact, boolean isPrivate) throws IllegalValueException {
+            this.isPrivate = isPrivate;
+            String trimmedcontact = contact.trim();
+            this.value = trimmedcontact;
+        }
+        
+        /**
+         * Test whether the string is a valid input.
+         * 
+         * @param test String
+         * @return whether the string matches the regular expression
+         */
+        
+        @Override
+        public boolean equals(Object other) {
```

Comment:
Note that this can be "equal" if the class we are comparing are Email class vs Address class, and their values are the same. Comparison should be done in the child classes for more detailed checks.